File: src/main/java/com/ggemo/va/bilidanmakuclient/http/MyHttpClient.java
package com.ggemo.va.bilidanmakuclient.http;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClientBuilder;
import org.apache.http.impl.client.HttpClients;
import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;

import java.io.IOException;

public class MyHttpClient {
    private CloseableHttpClient httpClient;
    private static final MyHttpClient INSTANCE = new MyHttpClient();

    private MyHttpClient() {
        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
        cm.setMaxTotal(100);
        HttpClientBuilder clientbuilder = HttpClients.custom().setConnectionManager(cm);
        clientbuilder.disableCookieManagement();
        this.httpClient = clientbuilder.build();
    }

    public static MyHttpClient getInstance() {
        return INSTANCE;
    }

    public CloseableHttpResponse request(HttpPost httpPost) throws IOException {
        return httpClient.execute(httpPost);
    }


    public CloseableHttpResponse request(HttpGet httpGet) throws IOException {
        return httpClient.execute(httpGet);
    }
}

File: blive-danmaku-client-jvm/src/main/kotlin/com/ggemo/va/blivedanmakuclient/inner/httpclient/MyHttpClient.kt
package com.ggemo.va.bilidanmakuclient.http

import org.apache.http.client.methods.CloseableHttpResponse
import org.apache.http.client.methods.HttpGet
import org.apache.http.client.methods.HttpPost
import org.apache.http.impl.client.CloseableHttpClient
import org.apache.http.impl.client.HttpClients
import org.apache.http.impl.conn.PoolingHttpClientConnectionManager
import java.io.IOException

class MyHttpClient private constructor() {
    private val httpClient: CloseableHttpClient
    @Throws(IOException::class)
    fun request(httpPost: HttpPost?): CloseableHttpResponse {
        return httpClient.execute(httpPost)
    }

    @Throws(IOException::class)
    fun request(httpGet: HttpGet?): CloseableHttpResponse {
        return httpClient.execute(httpGet)
    }

    companion object {
        val instance = MyHttpClient()
    }

    init {
        val cm = PoolingHttpClientConnectionManager()
        cm.maxTotal = 100
        val clientbuilder = HttpClients.custom().setConnectionManager(cm)
        clientbuilder.disableCookieManagement()
        httpClient = clientbuilder.build()
    }
}